src/ws/session.c

```diff
@@ -533,6 +537,45 @@ pam_conv_func_dummy (int num_msg,
   return PAM_CONV_ERR;
 }
 
+static void
+create_s4u_ticket (const char *username)
+{
+  OM_uint32 major, minor;
+  gss_buffer_desc user_buf = { .length = strlen (username), .value = (char *) username };
+  gss_cred_id_t server_cred = GSS_C_NO_CREDENTIAL;
+  gss_name_t impersonee = GSS_C_NO_NAME;
+
+  debug ("Attempting to create an S4U ticket for user %s", username);
+
+  if (!acquire_service_credentials (GSS_C_NO_OID, GSS_C_INITIATE, &server_cred))
+    goto out;
+
+  major = gss_import_name (&minor, &user_buf, GSS_KRB5_NT_PRINCIPAL_NAME, &impersonee);
+  if (GSS_ERROR (major))
+    {
+      warnx ("Failed to import user name %s: %s", username, gssapi_strerror (GSS_C_NO_OID, major, minor));
+      goto out;
+    }
+
+  /* store credentials into global creds; they will be put into the session and cleaned up in main() */
+  major = gss_acquire_cred_impersonate_name (&minor, server_cred, impersonee,
+                                             GSS_C_INDEFINITE, GSS_C_NO_OID_SET, GSS_C_INITIATE,
+                                             &creds, NULL, NULL);
```

Comment:
Another nag for a future cleanup: this global `creds` thing needs to die.  We should be passing that around explicitly.

Reply:
Agreed, this is a good follow-up thing. added to my TODO list.